sentry/src/main/java/io/sentry/ITransaction.java

```diff
@@ -79,6 +79,20 @@ public interface ITransaction extends ISpan {
   /** Schedules when transaction should be automatically finished. */
   void scheduleFinish();
 
+  /**
+   * Force finishes the transaction and it's child spans with the specified status. If the
+   * transaction is already finished this is a no-op.
+   *
+   * @param status The status to set the unfinished child spans / transaction to.
+   * @param forceKeep Ensures the transaction is kept even when it e.g. contains no child spans.
+   *     Useful for situations were the transaction and profile provide crucial context (e.g. ANRs)
+   */
+  @ApiStatus.Internal
+  void forceFinish(@NotNull final SpanStatus status, boolean forceKeep);
```

Comment:
tbqh this `forceFinish` + `forceKeep` made me confused 😅 would it make sense to change the naming a bit, something more explicit like `keepIfIdleChildless` maybe? Or we could invert it to `true` and rename to `dropChildless`. Your call

Reply:
yes this is awful! I even had the inverse variant in an early draft but couldn't find a right names for it. `dropChildless` sounds good, maybe `dropIfNoChildren` is even more readable?

Reply:
yeah `dropIfNoChildren` sounds good 👍